packages/gatsby/src/bootstrap/load-plugins/index.ts

```diff
@@ -36,11 +36,23 @@ const getAPI = (
 const flattenPlugins = (plugins: Array<IPluginInfo>): Array<IPluginInfo> => {
   const flattened: Array<IPluginInfo> = []
   const extractPlugins = (plugin: IPluginInfo): void => {
-    if (plugin.pluginOptions && plugin.pluginOptions.plugins) {
-      plugin.pluginOptions.plugins.forEach(subPlugin => {
-        flattened.push(subPlugin)
-        extractPlugins(subPlugin)
-      })
+    if (plugin.subPluginPaths) {
+      for (const subPluginPath of plugin.subPluginPaths) {
+        const segments = subPluginPath.split(`.`)
```

Comment:
CC @pieh

Reply:
`subPluginPath` can look like (for example) `someOption.randomFieldThatIsMarkedAsSubplugins`, so to traverse it we need to get segments and iterate on those.

Reason for doing stringified path with `.` separator was that it was just easier to prevent duplicates in `subPluginPaths` array (as each subplugin in the gatsby-config would add subplugin path.

Reply:
I've added a comment